intro-HPC/ch_gpu_gent.tex

```diff
@@ -0,0 +1,260 @@
+  or as a separate resource request (similar to the amount of memory) via \lstinline|-l gpus=Z|.
+  Both notations give exactly the same result.
+  The \lstinline|-l gpus=Z| is convenient is you only need one
+  node and you are fine with the default number of cores per GPU.
+  The \lstinline|-l nodes=...:gpus=Z| notation is required if you want to run with full control or in multinode cases like MPI jobs.
+  In you do not specify the number of GPUs, you get by default 1 GPU.
+\item As a resource of it's own, via \lstinline|--gpus X|.
+  In this case however, you are \emph{not} guaranteed that the GPUs are on the same node,
+  so your script or code must be able to deal with this.
+%  TODO We are providing a parallel wrapper ``wurker'' in the ``vsc-mympirun'' module to help with this
+%  (and with other more usual parallel work, similar to the usual ``worker'' or ``atools'' tools).
+%\item As a partial node resource, via ``-l nodes=...:mps=Z'' or ``-l mps=Z''.
+%  This triggers the Multi-Process Service (MPS, see https://docs.nvidia.com/deploy/pdf/CUDA_Multi_Process_Service_Overview.pdf),
+%  a way to ask for part of a GPU. The ``mps=`` value is a percentage of a GPU, and when submitting the job,
+%  you typically ask for a multiple of 100. The jobscript can then hand out portions of this (e.g. 50 per task) to the actual work.
+%  This is useful when a single application or MPI task cannot utilise a single/full GPU, and there are many other similar tasks that
+%  need to be processed or increasing the MPI ranks gives a speedup (e.g. when there is a significant portion of CPU work in the code).
+%  Unfortunately, this is not a silver bullet, and might require some experimenting to found out any potential benefits and proper tuning.
+%  TODO: how can a user now that an application is not using the full gpu resources?
+%  TODO this needs testing and there are some constraints (eg one mps job per node and thus one user per node using MPS)
+%  TODO needs proper integration with mypmirun / wurker
+%  TODO add separate section on MPS
+\end{itemize}
+
+Some background:
+\begin{itemize}
+\item The GPUs are constrained to the jobs (like the CPU cores), but do not run in so-called ``exclusive'' mode.
+\item The GPUs run with the so-called ``persistence daemon'', so the GPUs is not re-initialised between jobs.
+%  TODO: we need to fix this in the pro/epilogue scripts? is this similar to regular RAM?
+\end{itemize}
+
+
+%TODO: add examples ?
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+%%%%%%%%%%%%%%%%%%%%
+% Attention points %
+%%%%%%%%%%%%%%%%%%%%
+
+\section{Attention points}
+\label{sec:gpu_ugent_attention_points}
+
+Some important attention points:
+
+\begin{itemize}
+\item For MPI jobs, we recommend the (new) wrapper \lstinline|mypmirun| from the \lstinline|vsc-mympirun| module
```

Comment:
was there a specific version or it is part of the latest/default version?

Reply:
They should always load default/latest (there's only one version installed on `joltik` now, the `5.0.0-beta`)